### DISCUSSION TITLE
ENH: Cached Views

### DISCUSSION THREAD

**Discussion by pankajp** [duckdb/duckdb#2764] [General]
I'm writing an application which creates lots of views for different analysis which can be used by users, loading the data from parquet files. Some of the views are interdependent and expensive to compute. Is there a way to cache these views? The parquet files aren't expected to be changed, and I don't want to use materialized tables because not all the views would be needed in all cases and populating all of them would increase startup latency significantly.

What I would like is something like the behavior of Python's cached_property ( https://docs.python.org/3/library/functools.html#functools.cached_property ) decorator, to compute and store the view as a table the first time it is computed (maybe a cache per view column is better). We could have a syntax like `CREATE [CACHED] VIEW ...` or maybe some pragma.

I'm not sure if any other database provides this kind of functionality, but it seems pretty useful to me for any interactive analytics workload.

**Comment by danz68** [comment]
I've asked similar in a different thread. A materialized/cached view technique, especially for CTE, but not only can improve a lot, not only the speed but also the used memory. Search for CTE in Discussions and see how I used temporary tables as a surrogate for the missing cached/materialized views and how that improved my benchmarks (in a very specific situation, true, but you get the idea).
